keyboards/bpiphany/frosty_flake/frosty_flake.c

```diff
@@ -21,28 +32,16 @@ bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
     return process_record_user(keycode, record);
 }
 
-void led_set_kb(uint8_t usb_led) {
-    DDRB |= (1<<7);
-    DDRC |= (1<<5) | (1<<6);
-
-    print_dec(usb_led);
-
-    if (usb_led & (1<<USB_LED_CAPS_LOCK))
-        PORTC &= ~(1<<5);
-    else
-        PORTC |=  (1<<5);
-
-    if (usb_led & (1<<USB_LED_NUM_LOCK))
-        PORTB &= ~(1<<7);
-    else
-        PORTB |=  (1<<7);
+bool led_update_kb(led_t usb_led) {
+    // user requests no further processing
+    if (!led_update_user(usb_led))
+        return true;
 
-    if (usb_led & (1<<USB_LED_SCROLL_LOCK))
-        PORTC &= ~(1<<6);
-    else
-        PORTC |=  (1<<6);
+    writePin(C5, !usb_led.caps_lock);
+    writePin(B7, !usb_led.num_lock);
+    writePin(C6, !usb_led.scroll_lock);
 
-    led_set_user(usb_led);
+    return true;
 }
 
 __attribute__ ((weak))
```

Comment:
Everything from this line onwards can be deleted.

Reply:
Happy to. I kept most of these since they were in the original implementation, but I wasn’t sure if they were only valuable as a “here’s how to write a keyboard” template or had practical use.

Reply:
Done- I also pulled the two _above_ which were matrix_{init,scan}_kb and were also present in quantum/